src/main/java/uk/gov/mca/beacons/api/gateways/AuthGatewayImpl.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.mca.beacons.api.gateways;
+
+import com.azure.spring.aad.webapi.AADOAuth2AuthenticatedPrincipal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import uk.gov.mca.beacons.api.domain.User;
+
+@Component
+public class AuthGatewayImpl implements AuthGateway {
```

Comment:
The `Impl` suffix might be a [missed opportunity](http://www.kyleblaney.com/software-blog/2009/11/12/code-smell-classes-named-impl.html) to express something more meaningful about the implementation.  Could this implementation be named for what it gateways to?  E.g. `AzureAdAuthGateway`?

(Conscious this is the established pattern in the codebase!)